st2api/st2api/controllers/v1/action_views.py

```diff
@@ -185,10 +189,28 @@ def get_one(self, ref_or_id, requester_user):
             raise StackStormDBObjectNotFoundError('Action ref_or_id=%s has no entry_point to output'
                                                   % ref_or_id)
 
-        with open(abs_path) as file:
+        with open(abs_path, 'rb') as file:
             content = file.read()
 
-        return content
+        try:
+            content_type = mimetypes.guess_type(abs_path)[0]
+        except Exception:
+            content_type = None
+
+        # Special case if /etc/mime.types doesn't contain entry for yaml, py
+        if not content_type:
+            _, extension = os.path.splitext(abs_path)
+            if extension in ['.yaml', '.yml']:
```

Comment:
I added entries for "common cases" such as YAML and Python files.

Usually default ``/etc/mime.types`` file doesn't contain entry for YAML files so the endpoint would always just return ``text/plain`` which is not desired.